Mathlib/LinearAlgebra/LinearIndependent.lean

```diff
@@ -578,6 +585,14 @@ theorem LinearIndependent.units_smul {v : ι → M} (hv : LinearIndependent R v)
     rfl
 #align linear_independent.units_smul LinearIndependent.units_smul
 
+lemma LinearIndependent.eq_of_pair {x y : M} (h : LinearIndependent R ![x, y])
+    {s t s' t' : R} (h' : s • x + t • y = s' • x + t' • y) : s = s' ∧ t = t' := by
+  have : (s - s') • x + (t - t') • y = 0 := by
+    rw [← sub_eq_zero_of_eq h', ← sub_eq_zero]
+    simp [sub_smul, smul_smul]
```

Comment:
I note that this is followed by `abel` which should be fairly robust to minor changes in the goal but I think we should probably still avoid this non-terminal `simp`.